### ISSUE TITLE
Start plugin

### ISSUE THREAD

**Issue by overload08** [fritz-smh/domogik-plugin-weather#8]
2017-01-24 12:22:26,162 domogik-weather INFO Looking for launched instances of 'weather'
2017-01-24 12:22:26,174 domogik-weather INFO No existing process.
2017-01-24 12:22:26,174 domogik-weather DEBUG Setting locale to system locale: ('fr_FR', 'UTF-8')
2017-01-24 12:22:26,176 domogik-weather INFO ----------------------------------
2017-01-24 12:22:26,176 domogik-weather INFO Starting client 'weather' (new manager instance)
2017-01-24 12:22:26,176 domogik-weather INFO Python version is sys.version_info(major=2, minor=7, micro=9, releaselevel='final', serial=0)
2017-01-24 12:22:26,178 domogik-weather DEBUG Send client status : starting
2017-01-24 12:22:26,178 domogik-weather DEBUG Send client status : starting
2017-01-24 12:22:26,187 domogik-weather DEBUG Write pid file for pid '30623' in file '/var/run/domogik/weather.pid'
2017-01-24 12:22:26,188 domogik-weather INFO Read the json file and validate id
2017-01-24 12:22:26,193 domogik-weather INFO The json file is valid
2017-01-24 12:22:26,201 domogik-weather INFO Check if there are pictures for the defined products
2017-01-24 12:22:26,201 domogik-weather INFO There is no products defined for this client
2017-01-24 12:22:26,201 domogik-weather INFO End init of the global client part
2017-01-24 12:22:26,202 domogik-weather INFO Start of the xPL init
2017-01-24 12:22:26,202 domogik-weather INFO Configured interface(s) : [u'eth0']
2017-01-24 12:22:26,202 domogik-weather DEBUG Process informations|python_version=2.7|psutil_version=2.1.1|domogik_version=0.6.0|component=plugin-weather|component_version=1.7|pid=30623|cpu_percent_usage=0.0|memory_total=3018.0|memory_percent_usage=0.7|memory_rss=20.0|memory_vsz=337.0|num_threads=5|num_file_descriptors_used=17|platform=x86_64|num_core=2|git_branch=|git_revision=
2017-01-24 12:22:26,202 domogik-weather INFO The xpl connection may be done on the below addresses (only one will be used):
2017-01-24 12:22:26,203 domogik-weather INFO  - 10.0.101.126:0
2017-01-24 12:22:26,204 domogik-weather INFO Binding to ('10.0.101.126', 0) [BIND]...
2017-01-24 12:22:26,204 domogik-weather DEBUG xPL plugin weather socket bound to 10.0.101.126, port 33195
2017-01-24 12:22:26,205 domogik-weather INFO Plugin configure to have the xPL hub as mandatory!
2017-01-24 12:22:26,205 domogik-weather INFO HUB discovery > starting
2017-01-24 12:22:26,205 domogik-weather DEBUG send hbeat
2017-01-24 12:22:26,205 domogik-weather INFO HUB discovery > looking for the hub. I hope there is one hub, Domogik won't work without the hub!
HUB discovery > looking for the hub. I hope there is one hub, Domogik won't work without the hub!
2017-01-24 12:22:26,205 domogik-weather DEBUG normal send
2017-01-24 12:22:26,210 domogik-weather DEBUG xPL thread started for weather
2017-01-24 12:22:26,210 domogik-weather INFO HUB discovery > Received HBEAT echo, HUB detected
HUB discovery > Received HBEAT echo, HUB detected
2017-01-24 12:22:26,211 domogik-weather DEBUG New timer registered : <domogik.xpl.common.xplconnector.XplTimer instance at 0x7f66189058c0>
2017-01-24 12:22:26,211 domogik-weather DEBUG send hbeat
2017-01-24 12:22:26,211 domogik-weather INFO Try to register a thread twice: <Thread(thread-monitor, started 140076264716032)>
2017-01-24 12:22:26,211 domogik-weather INFO HUB discovery > hub found, configuration in progress
HUB discovery > hub found, configuration in progress
2017-01-24 12:22:26,212 domogik-weather DEBUG New timer created : <domogik.xpl.common.xplconnector.XplTimer instance at 0x7f66189058c0>
2017-01-24 12:22:26,212 domogik-weather DEBUG normal send
2017-01-24 12:22:26,212 domogik-weather DEBUG New listener 624bff74-e227-11e6-b3ef-00219bfc6edb, filter : {'xpltype': 'xpl-cmnd', 'schema': 'hbeat.request'}
2017-01-24 12:22:26,213 domogik-weather INFO End of the xPL init
2017-01-24 12:22:26,214 domogik-weather INFO Retrieve the devices list for this client...
2017-01-24 12:22:26,215 domogik-weather DEBUG send hbeat
2017-01-24 12:22:26,215 domogik-weather INFO HUB discovery > hub found, configuration in progress
HUB discovery > hub found, configuration in progress
2017-01-24 12:22:26,216 domogik-weather DEBUG normal send
2017-01-24 12:22:26,266 domogik-weather INFO - id : 14  /  name : Météo  /  device type id : weather.weather
2017-01-24 12:22:26,266 domogik-weather INFO   xpl_stats features :
2017-01-24 12:22:26,266 domogik-weather INFO   - current_wind_speed
2017-01-24 12:22:26,266 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,266 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,266 domogik-weather INFO     - type = speed
2017-01-24 12:22:26,266 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,266 domogik-weather INFO     - current
2017-01-24 12:22:26,266 domogik-weather INFO   - current_last_updated
2017-01-24 12:22:26,267 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,267 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,267 domogik-weather INFO     - type = last_updated
2017-01-24 12:22:26,267 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,267 domogik-weather INFO     - current
2017-01-24 12:22:26,267 domogik-weather INFO   - current_visibility
2017-01-24 12:22:26,267 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,267 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,267 domogik-weather INFO     - type = visibility
2017-01-24 12:22:26,267 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,267 domogik-weather INFO     - current
2017-01-24 12:22:26,267 domogik-weather INFO   - current_humidity
2017-01-24 12:22:26,268 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,268 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,268 domogik-weather INFO     - type = humidity
2017-01-24 12:22:26,268 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,268 domogik-weather INFO     - current
2017-01-24 12:22:26,268 domogik-weather INFO   - forecast_day_4
2017-01-24 12:22:26,268 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,268 domogik-weather INFO     - day = 4
2017-01-24 12:22:26,268 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,268 domogik-weather INFO     - provider = yahoo weather
2017-01-24 12:22:26,269 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,269 domogik-weather INFO     - condition-code
2017-01-24 12:22:26,269 domogik-weather INFO     - condition-text
2017-01-24 12:22:26,269 domogik-weather INFO     - day-name
2017-01-24 12:22:26,269 domogik-weather INFO     - temperature-high
2017-01-24 12:22:26,269 domogik-weather INFO     - temperature-low
2017-01-24 12:22:26,269 domogik-weather INFO   - current_feels_like
2017-01-24 12:22:26,269 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,269 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,269 domogik-weather INFO     - type = temp_feels_like
2017-01-24 12:22:26,270 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,270 domogik-weather INFO     - current
2017-01-24 12:22:26,270 domogik-weather INFO   - current_wind_direction
2017-01-24 12:22:26,270 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,270 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,270 domogik-weather INFO     - type = direction
2017-01-24 12:22:26,270 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,270 domogik-weather INFO     - current
2017-01-24 12:22:26,270 domogik-weather INFO   - current_temperature
2017-01-24 12:22:26,270 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,270 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,271 domogik-weather INFO     - type = temp
2017-01-24 12:22:26,271 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,271 domogik-weather INFO     - current
2017-01-24 12:22:26,271 domogik-weather INFO   - current_sunrise
2017-01-24 12:22:26,271 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,271 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,271 domogik-weather INFO     - type = sunrise
2017-01-24 12:22:26,271 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,271 domogik-weather INFO     - current
2017-01-24 12:22:26,271 domogik-weather INFO   - current_text
2017-01-24 12:22:26,272 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,272 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,272 domogik-weather INFO     - type = text
2017-01-24 12:22:26,272 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,272 domogik-weather INFO     - current
2017-01-24 12:22:26,272 domogik-weather INFO   - forecast_day_3
2017-01-24 12:22:26,272 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,272 domogik-weather INFO     - day = 3
2017-01-24 12:22:26,272 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,273 domogik-weather INFO     - provider = yahoo weather
2017-01-24 12:22:26,273 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,273 domogik-weather INFO     - condition-code
2017-01-24 12:22:26,273 domogik-weather INFO     - condition-text
2017-01-24 12:22:26,273 domogik-weather INFO     - day-name
2017-01-24 12:22:26,273 domogik-weather INFO     - temperature-high
2017-01-24 12:22:26,273 domogik-weather INFO     - temperature-low
2017-01-24 12:22:26,273 domogik-weather INFO   - forecast_day_2
2017-01-24 12:22:26,273 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,273 domogik-weather INFO     - day = 2
2017-01-24 12:22:26,273 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,274 domogik-weather INFO     - provider = yahoo weather
2017-01-24 12:22:26,274 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,274 domogik-weather INFO     - condition-code
2017-01-24 12:22:26,274 domogik-weather INFO     - condition-text
2017-01-24 12:22:26,274 domogik-weather INFO     - day-name
2017-01-24 12:22:26,274 domogik-weather INFO     - temperature-high
2017-01-24 12:22:26,274 domogik-weather INFO     - temperature-low
2017-01-24 12:22:26,274 domogik-weather INFO   - forecast_day_1
2017-01-24 12:22:26,274 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,275 domogik-weather INFO     - day = 1
2017-01-24 12:22:26,275 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,275 domogik-weather INFO     - provider = yahoo weather
2017-01-24 12:22:26,275 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,275 domogik-weather INFO     - condition-code
2017-01-24 12:22:26,275 domogik-weather INFO     - condition-text
2017-01-24 12:22:26,275 domogik-weather INFO     - day-name
2017-01-24 12:22:26,275 domogik-weather INFO     - temperature-high
2017-01-24 12:22:26,275 domogik-weather INFO     - temperature-low
2017-01-24 12:22:26,275 domogik-weather INFO   - forecast_day_0
2017-01-24 12:22:26,276 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,276 domogik-weather INFO     - day = 0
2017-01-24 12:22:26,276 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,276 domogik-weather INFO     - provider = yahoo weather
2017-01-24 12:22:26,276 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,276 domogik-weather INFO     - condition-code
2017-01-24 12:22:26,276 domogik-weather INFO     - condition-text
2017-01-24 12:22:26,276 domogik-weather INFO     - day-name
2017-01-24 12:22:26,276 domogik-weather INFO     - temperature-high
2017-01-24 12:22:26,276 domogik-weather INFO     - temperature-low
2017-01-24 12:22:26,277 domogik-weather INFO   - current_station
2017-01-24 12:22:26,277 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,277 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,277 domogik-weather INFO     - type = current_station
2017-01-24 12:22:26,277 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,277 domogik-weather INFO     - current
2017-01-24 12:22:26,277 domogik-weather INFO   - current_sunset
2017-01-24 12:22:26,277 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,277 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,277 domogik-weather INFO     - type = sunset
2017-01-24 12:22:26,278 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,278 domogik-weather INFO     - current
2017-01-24 12:22:26,278 domogik-weather INFO   - current_code
2017-01-24 12:22:26,278 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,278 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,278 domogik-weather INFO     - type = code
2017-01-24 12:22:26,278 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,278 domogik-weather INFO     - current
2017-01-24 12:22:26,278 domogik-weather INFO   - current_barometer_value
2017-01-24 12:22:26,278 domogik-weather INFO     Static Parameters :
2017-01-24 12:22:26,279 domogik-weather INFO     - device = 583959
2017-01-24 12:22:26,279 domogik-weather INFO     - type = pressure
2017-01-24 12:22:26,279 domogik-weather INFO     Dynamic Parameters :
2017-01-24 12:22:26,279 domogik-weather INFO     - current
2017-01-24 12:22:26,279 domogik-weather INFO   xpl_commands features :
2017-01-24 12:22:26,279 domogik-weather INFO Try to register a thread twice: <Thread(thread-monitor, started 140076264716032)>
2017-01-24 12:22:26,279 domogik-weather DEBUG Get parameter 'device' for 'xpl_stats', feature 'current_temperature'
2017-01-24 12:22:26,279 domogik-weather DEBUG send hbeat
2017-01-24 12:22:26,280 domogik-weather DEBUG Parameter value found: 583959
2017-01-24 12:22:26,280 domogik-weather DEBUG normal send
2017-01-24 12:22:26,280 domogik-weather INFO Start getting weather for Météo (583959)
2017-01-24 12:22:26,282 domogik-weather DEBUG Send client status : alive
2017-01-24 12:22:26,282 domogik-weather DEBUG Url called is https://query.yahooapis.com/v1/public/yql?q=select * from weather.forecast where woeid = 583959 and u = 'f'&format=json
2017-01-24 12:22:26,282 domogik-weather INFO Start IOLoop for MQ : nothing else can be executed in the __init__ after this! Make sure that the self.ready() call is the last line of your init!!!!
2017-01-24 12:22:31,969 domogik-weather DEBUG Raw data for 583959 : {u'query': {u'count': 0, u'lang': u'en-US', u'results': None, u'created': u'2017-01-24T11:22:45Z'}}
2017-01-24 12:22:31,970 domogik-weather ERROR Error while getting data from Yahoo weather : Traceback (most recent call last):
  File "/var/lib/domogik/domogik_packages/plugin_weather/lib/weather.py", line 114, in get_weather
    cur = data['query']['results']['channel']
TypeError: 'NoneType' object has no attribute '__getitem__'

2017-01-24 12:22:31,970 domogik-weather INFO Wait for 15 minutes

**Comment by overload08** [comment]
May be duplicate of #7

**Comment by overload08** [comment]
```
diff --git a/lib/weather.py b/lib/weather.py
index 3a9c28e..298acda 100644
--- a/lib/weather.py
+++ b/lib/weather.py
@@ -108,7 +108,9 @@ class Weather:
                 if 'error' in data:
                     self.log.error("Error raised by Yahoo weather : {0}".format(data['error']))
                     return
-
+                if data['query']['results'] == None:
+                    self.log.error("Error raised by Yahoo weather : Service unavaillable")
+                    return

                 ### send current data over xPL
                 cur = data['query']['results']['channel']
```

**Comment by fritz-smh** [comment]
Fixed in 1.8
